backend/btrixcloud/operator.py

```diff
@@ -177,19 +178,29 @@ async def sync_crawls(self, data: MCSyncData):
 
         redis_url = self.get_redis_url(crawl_id)
 
+        params = {}
+        params.update(self.shared_params)
+        params["id"] = crawl_id
+        params["cid"] = cid
+        params["userid"] = spec.get("userid", "")
+
         # if finalizing, crawl is being deleted
         if data.finalizing:
-            # if not yet finished, assume it was canceled, mark as such
-            print(f"Finalizing crawl {crawl_id}, finished {status.finished}")
             if not status.finished:
-                await self.cancel_crawl(redis_url, crawl_id, cid, status, "canceled")
+                # if can't cancel, already finished
+                if not await self.cancel_crawl(
+                    redis_url, crawl_id, cid, status, "canceled"
+                ):
+                    # instead of fetching the state (that was already set)
+                    # return exception to ignore this request, keep previous
+                    # finished state
+                    raise HTTPException(status_code=400, detail="out of sync status")
```

Comment:
```suggestion
                    raise HTTPException(status_code=400, detail="out_of_sync_status")
```

Just to be more consistent with other HTTPException details we're sending